Google-Summer-of-Code-2023.md

```diff
@@ -273,3 +273,504 @@ The following is a list of Oppia's 2023 GSoC project ideas. You are welcome to c
+
+**Proposal notes:**
+
+* We’ll likely write a beam job to populate the auto-translations for a language. See the [wiki](https://github.com/oppia/oppia/wiki/Apache-Beam-Jobs) for more details. 
+* Maybe ~80% of the CAT work is already done, including all of the backend changes. We just need to hook it up in the frontend, e.g. see working PR: [#12624](https://github.com/oppia/oppia/pull/12604/files) and related issue [#11434](https://github.com/oppia/oppia/issues/11434). 
+  - See Computer Aided Translations (CAT) backend milestone PRs.
+* The CAT backend utilizes the Google Translate API for auto-translations. We can re-use that work here for auto-translations. Alternatively, we can investigate other translate APIs that may offer better translations, e.g. Oppia has done studies in the past that have shown Microsoft to produce preferred translations.
+
+
+**Useful resources:**
+
+### 2.3. Improvements for translation reviewer experience
+
+**Project Description:**
+
+Translation reviewers have given feedback such that the current contributor dashboard reviewer workflow doesn’t match with their most of the needs. This project aims to improve the translation reviewer experience in the contributor dashboard.
+**Size of this project:** large (~350 hours)
+
+**Potential Mentors:** 
+
+**Knowledge/Skills Recommended:**
+
+* Knowledge and understanding of Python
+* Knowledge and understanding of TypeScript, Angular, HTML, and CSS
+
+**Suggested Milestones:**
+
+* **Milestone 1:**
+  - Showing translation cards according to the order of the content flow in the lesson. Exploration editor should also be linked so that the original lesson can be seen.
+  - Showing the size of the translation(short/large)
+  - Pin/unpin specific lessons to translate
+
+
+* **Milestone 2:**
+  - Ability to revert the acceptance or rejection of cards. The exact time limit that we allow to undo the review should be decided.
+  - Showing image alt text below the image
+  - Notifying new submissions through emails
+
+
+**Dependency on Release Schedule:** 
+
+**Proposal notes:**
+
+* To support lesson pinning, we have to make sure the pinned lessons are returned at the top of the list of lessons with translation suggestions in the fetch query [here](https://github.com/oppia/oppia/blob/develop/core/controllers/contributor_dashboard.py#L336).
+* We will likely want to introduce a new storage model to track pinned lessons by language and user.
+
+
+**Useful resources:**
+
+## Server data validation team
+
+### 3.1. Fix Validation Errors
+
+TBA
+
+## Developer workflow team
+
+### 4.1. Dockerize Oppia
+
+**Project Description:**
+
+We want to package Oppia as a Docker container so that new contributors can get started by running docker-compose up -d after cloning the repository to install all the dependencies and start the local development server. This will be much easier than the current installation process, which is often painful to troubleshoot.
```

Comment:
```suggestion
We want to package Oppia as a Docker container so that new contributors can get started by running `docker-compose up -d` after cloning the repository to install all the dependencies and start the local development server. This will be much easier than the current installation process, which is often painful to troubleshoot. Docker will also let us speed up test runs on GitHub Actions by caching the Docker image.
```

Added the note about GitHub Actions to address this comment on the idea doc: https://docs.google.com/document/d/1DnNy87Vhpa6748RzvsDKtAfzyl8DRRexPTFPVhFxYXU/edit?disco=AAAAphfDuHY